File: backend/AgPolicy.Api/Models/Quote.cs
namespace AgPolicy.Api.Models;

public class Quote
{
    public int Id { get; set; }
    public int FarmerId { get; set; }
    public Farmer? Farmer { get; set; }
    public int FarmId { get; set; }
    public Farm? Farm { get; set; }
    public string CropType { get; set; } = string.Empty;
    public decimal Acres { get; set; }
    public int CoverageLevel { get; set; }
    public decimal EstimatedPremium { get; set; }
    public string Status { get; set; } = "Draft";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public int? ConvertedPolicyId { get; set; }
}

File: frontend/agpolicy-client/src/types/quote.ts
export interface Quote {
  id: number
  farmerId: number
  farmerName: string
  farmId: number
  farmName: string
  cropType: string
  acres: number
  coverageLevel: number
  estimatedPremium: number
  status: string
  createdAt: string
  convertedPolicyId?: number | null
}

export interface CreateQuoteRequest {
  farmerId: number
  farmId: number
  cropType: string
  acres?: number
  coverageLevel: number
}

export interface ConvertQuoteResponse {
  quoteId: number
  policyId: number
  message: string
}
